druid/src/text/rich_text.rs

```diff
@@ -131,6 +135,17 @@ impl RichTextBuilder {
         self.add_attributes_for_range(range)
     }
 
+    /// Glue for usage of the write! macro.
+    ///
+    /// This method should generally not be invoked manually, but rather through the write! macro itself.
+    pub fn write_fmt(&mut self, fmt: std::fmt::Arguments<'_>) -> AttributesAdder {
```

Comment:
Maybe add `#[doc(hidden)]` on this, since it's really an implementation detail and doesn't need to show up in public docs?

Hmm, should we be implementing the `Write` trait directly? Is this depending on an implementation detail of the macro, where it calls `write_fmt` with no qualifiers, instead of doing like `Write::write_fmt(writer, args)`?

Reply:
I didn't implement `Write` because it returns a Result, I wanted it to feel more like `format!`.

> Is this depending on an implementation detail of the macro?

the [docs](https://doc.rust-lang.org/stable/std/macro.write.html) say it just needs something with `write_fmt` method.

> The writer may be any value with a write_fmt method; generally this comes from an implementation of either the fmt::Write or the io::Write trait.